spring-integration-cloudevents/src/main/java/org/springframework/integration/cloudevents/transformer/ToCloudEventTransformer.java

```diff
@@ -0,0 +1,415 @@
+import io.cloudevents.rw.CloudEventWriter;
+import io.cloudevents.rw.CloudEventWriterFactory;
+import org.jspecify.annotations.Nullable;
+
+import org.springframework.context.ApplicationContext;
+import org.springframework.expression.EvaluationContext;
+import org.springframework.expression.Expression;
+import org.springframework.expression.common.LiteralExpression;
+import org.springframework.integration.expression.ExpressionUtils;
+import org.springframework.integration.expression.FunctionExpression;
+import org.springframework.integration.expression.ValueExpression;
+import org.springframework.integration.support.utils.PatternMatchUtils;
+import org.springframework.integration.transformer.AbstractTransformer;
+import org.springframework.integration.transformer.MessageTransformationException;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHeaders;
+import org.springframework.messaging.support.MessageBuilder;
+import org.springframework.messaging.support.MessageHeaderAccessor;
+import org.springframework.util.Assert;
+import org.springframework.util.MimeType;
+
+/**
+ * Convert messages to CloudEvent format.
+ * Perform attribute and extension mapping based on {@link Expression}s.
+ *
+ * @author Glenn Renfro
+ * @since 7.1
+ */
+public class ToCloudEventTransformer extends AbstractTransformer {
+
+	private static final String DEFAULT_PREFIX = "ce-";
+
+	private static final EventFormatProvider eventFormatProvider = EventFormatProvider.getInstance();
+
+	private final String[] extensionPatterns;
+
+	private String cloudEventPrefix = DEFAULT_PREFIX;
+
+	private boolean failOnNoFormat = false;
+
+	private Expression eventIdExpression = new FunctionExpression<Message<?>>(
+			msg -> Objects.requireNonNull(msg.getHeaders().getId()).toString());
+
+	@SuppressWarnings("NullAway.Init")
+	private Expression sourceExpression;
+
+	private Expression typeExpression = new LiteralExpression("spring.message");
+
+	@SuppressWarnings("NullAway.Init")
+	private @Nullable Expression dataSchemaExpression;
+
+	private @Nullable Expression subjectExpression;
+
+	@SuppressWarnings("NullAway.Init")
+	private EvaluationContext evaluationContext;
+
+	/**
+	 * Create a ToCloudEventTransformer with no extensionPatterns.
+	 * @since 7.1
+	 */
+	public ToCloudEventTransformer() {
+		this(new String[0]);
+	}
+
+	/**
+	 * Create a ToCloudEventTransformer.
+	 * @param extensionPatterns patterns to evaluate whether message headers should be added as extensions
+	 *                          to the CloudEvent
+	 * @since 7.1
+	 */
+	public ToCloudEventTransformer(String... extensionPatterns) {
+		this.extensionPatterns = Arrays.copyOf(extensionPatterns, extensionPatterns.length);
+	}
+
+	@Override
+	protected void onInit() {
+		super.onInit();
+		this.evaluationContext = ExpressionUtils.createStandardEvaluationContext(getBeanFactory());
+		if (this.sourceExpression == null) {  // in the case the user sets the value prior to onInit.
```

Comment:
Please, remove the comment.
It does not give any value. 
Plus, it sounds like does not reflect reality: the condition is about `null`, but sentence in comment says something opposite.